CLAUDE.md

```diff
@@ -0,0 +1,216 @@
+- `agent.*` - Core agent tasks
+- `test` - Testing tasks
+- `linter.*` - Linting tasks
+- `docker.*` - Docker image tasks
+- `release.*` - Release management
+
+### Build Tags
+Go build tags control feature inclusion:
+- `kubeapiserver` - Kubernetes API server support
+- `containerd` - containerd support
+- `docker` - Docker support
+- `ebpf` - eBPF support
+- `python` - Python check support
+
+## Important Files
+
+### Configuration
+- `datadog.yaml` - Main agent configuration
+- `modules.yml` - Go module definitions
+- `release.json` - Release version information
+- `.gitlab-ci.yml` - CI/CD pipeline configuration
+
+### Documentation
+- `/docs/` - Internal documentation
+- `/docs/dev/` - Developer guides
+- `README.md` - Project overview
+- `CONTRIBUTING.md` - Contribution guidelines
+
+## CI/CD Pipeline
+
+### GitLab CI
+- Primary CI system
+- Defined in `.gitlab-ci.yml` and `.gitlab/` directory
+- Runs tests, builds, and deployments
+
+### GitHub Actions
+- Secondary CI for specific workflows
+- Windows unit tests
+- Code coverage reporting
+
+## Security Considerations
+
+### Sensitive Data
+- Never commit API keys or secrets
+- Use secret backend for credentials
+- Scrub sensitive data from logs
+
+### Code Paths
+- `/pkg/security/` - Security monitoring code
+- `/pkg/compliance/` - Compliance checking
+- eBPF code is GPL-licensed (separate from Apache 2.0)
+
+## Debugging Tips
+
+1. **Enable debug logging**: Set `log_level: debug` in config
+2. **Check agent status**: `./bin/agent/agent status`
+3. **Inspect flare**: `./bin/agent/agent flare`
+4. **View check output**: `./bin/agent/agent check <check_name>`
+
+## Module System
+The project uses Go modules with multiple sub-modules. Key modules:
+- Main module: `github.com/DataDog/datadog-agent`
+- Sub-modules defined in `modules.yml`
+- Use `dda inv modules.show` to list all modules
+
+## Platform Support
+- **Linux**: Full support (amd64, arm64)
+- **Windows**: Full support (Server 2016+, Windows 10+)
+- **macOS**: Development and testing
+- **AIX**: No support in this codebase
+- **Container**: Docker, Kubernetes, ECS, containerd, and more
+
+## Best Practices
+
+1. **Always run linters before committing**: `dda inv linter.go`
+2. **Test your changes**: `dda inv test --targets=<your_package>`
+3. **Follow Go conventions**: Use gofmt, follow project structure
+4. **Update documentation**: Keep docs in sync with code changes
+5. **Use meaningful commit messages**: Follow conventional commits
+6. **Check for security implications**: Review security-sensitive changes carefully
+
+## Troubleshooting Development Issues
+
+### Common Build Issues
+- **Missing tools**: Run `dda inv install-tools`
+- **CMake errors**: Remove `dda inv rtloader.clean`
+
+### Testing Issues
+- **Flaky tests**: Check `flakes.yaml` for known issues
+- **Integration test failures**: Ensure services are running
+- **Coverage issues**: Use `--coverage` flag
+
+## Additional Resources
```

Comment:
I would recommend adding the public developer docs site because we're going to heavily improve that shortly and continue removing one off pages in the docs directory. https://datadoghq.dev/datadog-agent/